specs/ingress/tests.go

```diff
@@ -58,50 +59,219 @@ func getExternalIP(svc, ns string) (string, error) {
 	return strings.Trim(ip, "'"), nil
 }
 
-func testNginx() {
+func testSampleAppInstall(ingressName string) {
 	h, _ := utils.GetHelperAndConfig()
-	ginkgo.By("Creating ingress-nginx controller")
-	_, err := h.Kubectl("", "apply", "-f", "testdata/ingress/controllers/nginx.yaml")
 
-	gomega.Expect(err).Should(gomega.BeNil(), fmt.Sprintf("failed to create controller: %s", utils.Err(err)))
+	switch ingressName {
+	case gloo:
+		// Install and inject booksapp
+		utils.TestBooksappApp()
```

Comment:
Hmm :thinking: Why not use a single sample application (could be emojivoto or a even smaller one) for all the tests, so that we have consistent experience and less manifest code?

Reply:
Shouldn't be a problem. I just wanted to replicate the tests as they're on the [docs](https://linkerd.io/2/tasks/using-ingress/index.html)
That way I wouldn't have to deal with changing any of the ingress resource configuration as instructions the docs are known to not cause any trouble. :)

Reply:
I agree with @Pothulapati that we should use only one sample application. If extra work and time are needed to figure out the configuration needed for gloo and contour, we can tackle them in separate PRs. That way, I can still merge this PR that covers most of the other ingress controllers.